src/main/kotlin/MinecraftSettings.kt

```diff
@@ -37,6 +37,9 @@ class MinecraftSettings : PersistentStateComponent<MinecraftSettings.State> {
         var underlineType: UnderlineType = UnderlineType.DOTTED,
 
         var isShadowAnnotationsSameLine: Boolean = true,
+
+        var isForceJsonTranslationFile: Boolean = false,
+        var convertToTranslationTemplate: String = "net.minecraft.client.resources.I18n.format(\"\$key\")",
```

Comment:
I believe intentions can have their own settings, so no need to make this a global setting. Although in this case we could probably get away with it being hardcoded if you use `MappingsManager`. Right now you would need to add it to `HardcodedYarnToMojmap` as we haven't fully implemented the mappings API yet.

It would be nice to figure out why we're unable to detect the MC version rather than forcing it. But this is probably good enough for a temporary solution.

Reply:
Its not just about mapping from yarn to mojmap, I personally like to use a small wrapper that can do some additional formatting. People on other platforms (paper, etc.) might also want to leverage the intention without being forced to use NMS. If you still want it to go through HardcodedYarnToMojmap let me know and I'll make the change

I'll look into how to move the settings into the intention settings

Reply:
That's a fair enough reason, although I think the usage of yarn is common enough that we need to be able to make it work without any config changes. I guess just leave the remapping problem for now and we'll fix it once we get the mappings stuff up and running.

Reply:
I'll try see if I can add an explicit checkbox for "Use Custom Template .. etc", which should make it easier to implement that behaviour in the future.

Reply:
It was found that intentions cannot have their own settings. Leaving the options in MinecraftConfigurable seems like the easiest solution right now